test/framework/dump.go

```diff
@@ -173,16 +173,28 @@ func (f *CommonFramework) DumpLogsForPodsWithLabelsInNamespace(ctx context.Conte
 
 	var result error
 	for _, pod := range pods.Items {
-		if err := f.DumpLogsForPodInNamespace(ctx, k8sClient, namespace, pod.Name, &corev1.PodLogOptions{}); err != nil {
-			result = multierror.Append(result, err)
+		for _, container := range pod.Spec.InitContainers {
+			if err := f.DumpLogsForPodInNamespace(ctx, k8sClient, namespace, pod.Name, &corev1.PodLogOptions{Container: container.Name}); err != nil {
+				result = multierror.Append(result, fmt.Errorf("error reading logs from pod %q init container %q: %w", pod.Name, container.Name, err))
+			}
+		}
+		for _, container := range pod.Spec.Containers {
+			if err := f.DumpLogsForPodInNamespace(ctx, k8sClient, namespace, pod.Name, &corev1.PodLogOptions{Container: container.Name}); err != nil {
+				result = multierror.Append(result, fmt.Errorf("error reading logs from pod %q container %q: %w", pod.Name, container.Name, err))
+			}
 		}
 	}
 	return result
 }
 
 // DumpLogsForPodInNamespace prints the logs of the pod with the given namespace and name.
 func (f *CommonFramework) DumpLogsForPodInNamespace(ctx context.Context, k8sClient kubernetes.Interface, namespace, name string, options *corev1.PodLogOptions) error {
-	log := f.Logger.WithValues("pod", client.ObjectKey{Namespace: namespace, Name: name})
+	var log logr.Logger
+	if options != nil {
+		log = f.Logger.WithValues("pod", client.ObjectKey{Namespace: namespace, Name: name}, "container", options.Container)
+	} else {
+		log = f.Logger.WithValues("pod", client.ObjectKey{Namespace: namespace, Name: name})
+	}
```

Comment:
```suggestion
	log := f.Logger.WithValues("pod", client.ObjectKey{Namespace: namespace, Name: name})
	if options != nil && options.Container != "" {
		log = log.WithValues("container", options.Container)
	}
```

Nit, a bit more concise.